### PR TITLE
Inline modules in `app/ide-desktop/`

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Remove unnecessary modules
  - Remove `ts-plugin-namespace-auto-import` as it was a workaround to use the non-conventional `import *` convention
  - Remove `esbuild-plugin-copy-directories` as it is unuse
- Inline modules that are only ever used once
  - Inline `project-manager-shim` into `gui2` - it is only used during `gui2`'s dev mode
  - Inline `content-config` into `client`
- Flatten `app/ide-desktop/lib/` to `app/ide-desktop/`
- Flatten `app/ide-desktop/lib/dashboard/` to `app/dashboard/`

### Important Notes
- As mentioned above, all remaining modules have been moved up from `app/ide-desktop/lib/` to `app/ide-desktop/`. It's not ideal but I'd rather hold off on moving them anywhere else before we have a consensus on what should go where.
  - (That is to say, this may not be the final directory structure - but I figure it's fine to get *something* done so that hopefully the rest of the restructuring is simpler.)

### Code Review Notes
- There are a lot of files changed, but the vast majority are renames without changes and hence have no need to be reviewed.
- Similarly, there are a lot of lines changed, but for the most part that is unavoidable due to the flattening significantly changing the order of entries in the `package.json`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

**Review by MrFlashAccount — APPROVED** [review]

**Comment by farmaazon** [comment]
> Flatten app/ide-desktop/lib/ to app/

Well, when the directory structure was designed, I imagined `app` folder representing not _the application_ but rather _all applications_:
* engine (compiler and maybe project manager)
* GUI (a package common for cloud browser and electron package)
* ide-desktop - electron package using GUI above

Thence came gui and ide-desktop. Much later on, dashboard landed inside `ide-desktop` which started breaking the above schema (it should land in GUI instead... but GUI was then in rust). Also, engine never made it to this `app` folder, because none wanted to spend time on sbt reconfiguring.

But still, I think we should ultimately keep separation between GUI and electron, just for cloud-in-browser sake.

~~Also, **these changes may greatly conflict with @Frizi efforts of migrating to pnpm - a part of moving to Bazel**, mostly done as far as I'm aware. I would wait for @Frizi changes to land first, to avoid nasty conflicts. Also, all significant directory restructuring should be consulted with "Bazel" team (@hubertp and @Frizi)~~

Edit: ok, I think I'm not aware, or misremembered some arrangements, so please ignore the second part.

File: app/gui2/projectManagement.ts
Comment (farmaazon) [inline app/gui2/projectManagement.ts]: Judging by this documentation, I don't know why it is in gui2 directory. gui2 is essentially `Project View` and the project view is not responsible for importing or managing projects through PM.

Please, add some TODO that this is required only for dev build (which should be soon reworked anyway).

File: app/gui2/projectManagerShimMiddleware.ts
Comment (farmaazon) [inline app/gui2/projectManagerShimMiddleware.ts]: This docs seem to be plainly wrong.

**Review by farmaazon — COMMENTED** [review]
For now, I would keep at least electron-specific packages still in `ide-desktop` (removing unnecessary `lib`)

**Comment by somebody1234** [comment]
umm...
https://github.com/enso-org/enso/actions/runs/9675920069/job/26694401507#step:7:639

anyone know how to fix project references in TS?

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Inline modules in `app/ide-desktop/` · enso-org/enso@0c52059</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by somebody1234** [comment]
Pinging @Frizi for visibility

File: app/ide-desktop/client/src/preload.ts
Comment (Frizi) [inline app/ide-desktop/client/src/preload.ts]: should this be a type-only import?

File: app/dashboard/src/components/AriaComponents/Form/components/useForm.ts
Comment (Frizi) [inline app/dashboard/src/components/AriaComponents/Form/components/useForm.ts]: Why is this change necessary?

File: app/dashboard/vite.config.ts
Comment (Frizi) [inline app/dashboard/vite.config.ts]: This version mismatch is now fixed in develop, those expect-errors will probably be flagged after merge.

File: app/gui2/projectManagement.ts
Comment (Frizi) [inline app/gui2/projectManagement.ts:1]: Lets move all `project-manager-shim`-related files to a subdiriectory.

File: build/build/src/ide/web.rs
Comment (Frizi) [inline build/build/src/ide/web.rs]: There is no `app/client`, should there be `ide-desktop/client`?

File: app/eslint.config.mjs
Comment (Frizi) [inline app/eslint.config.mjs:1]: I'm not a fan of this eslint config leaking into other apps now, but I guess it filters the files to dashboard anyway. We will have to tidy it up (i.e. remove most of the coutnerproductive rules) in a separate PR.

**Review by Frizi — APPROVED** [review]
Spotted some minor things to check or fix, but overall that this is a very good change and will simplify a bunch of things in the future 👍

**Comment by somebody1234** [comment]
@jdunkerley i think we need the approval of someone from libs because `dataLinkSchema.json` is being moved.

also the CODEOWNERS file has been updated and that one should probably be reviewed too

File: app/dashboard/src/components/AriaComponents/Form/components/useForm.ts
Comment (somebody1234) [reply app/dashboard/src/components/AriaComponents/Form/components/useForm.ts]: no clue :skull: - i guess @MrFlashAccount will know more about this

**Comment by somebody1234** [comment]
@farmaazon @Frizi either of you happen to know why lint CI is saying this?
https://github.com/enso-org/enso/actions/runs/9714248920/job/26812883079?pr=10305#step:7:200
```
 INFO ide_ci::program::command: npm⚠️ npm ERR! code 127
 INFO ide_ci::program::command: npm⚠️ npm ERR! path /runner/_work/enso/enso/app/gui2
 INFO ide_ci::program::command: npm⚠️ npm ERR! command failed
 INFO ide_ci::program::command: npm⚠️ npm ERR! command sh -c playwright install
 INFO ide_ci::program::command: npm⚠️ npm ERR! sh: 1: playwright: not found
```

i also got it while running locally but not sure what could possibly cause this...

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Inline modules in `app/ide-desktop/` · enso-org/enso@cc9493e</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by farmaazon** [comment]
I cannot reproduce this locally: I only have different errors in tests.

It looks really strange: the only place where I see `playwright install` is in our postinstall, but there it should be run from local package, so playwright should be visible.

**Comment by somebody1234** [comment]
hmm, i think it's only reproducible if you're missing certain packages, not sure which. maybe it might be reproducible if you delete `node_modules`?

> there it should be run from local package

yeah, indeed... no idea how it could possibly fail tbh. which is why i just ignored it and moved on when i encountered it

**Comment by farmaazon** [comment]
You may check what if you add `npx` there.

After removing node modules, I get a different error:

```
 INFO ide_ci::program::command: npmℹ️ src/entrypoint.ts(4,28): error TS6305: Output file '/home/adam-praca/dev/enso/node_modules/.cache/tsc/enso/app/dashboard/src/index.d.ts' has not been built from source file '/home/adam-praca/dev/enso/app/dashboard/src/index.tsx'.
 INFO ide_ci::program::command: npm⚠️ npm ERR! Lifecycle script `typecheck` failed with error: 
 INFO ide_ci::program::command: npm⚠️ npm ERR! Error: command failed 
 INFO ide_ci::program::command: npm⚠️ npm ERR!   in workspace: enso-gui2@0.1.0 
 INFO ide_ci::program::command: npm⚠️ npm ERR!   at location: /home/adam-praca/dev/enso/app/gui2 
```

File: app/dashboard/src/components/AriaComponents/Form/components/useForm.ts
Comment (MrFlashAccount) [reply app/dashboard/src/components/AriaComponents/Form/components/useForm.ts]: @somebody1234 but this change doesn't exist on develop... not sure I can explain it either...

Comment (somebody1234) [reply app/dashboard/src/components/AriaComponents/Form/components/useForm.ts]: yeah i have no idea XD it's probably to do with some accidental version bump, i guess i'll take a look on monday

**Comment by somebody1234** [comment]
idk what's up with CI o_O
```
Cannot find package 'yaml' imported from /runner/_work/enso/enso/app/gui2/vitest.config.ts.timestamp-1720431019684-09bd4b9a42eb1.mjs
```

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**
